modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/PostgisPSFilterToSql.java

```diff
@@ -187,6 +189,8 @@ public Object visit(PropertyIsEqualTo filter, Object extraData) {
                                     throw new RuntimeException(e);
                                 }
                             }));
+        } else if (any != null) {
+            return helper.visit(any, extraData);
```

Comment:
Feedback: A question that goes beyond this PR review - is the duplication necessary? Can one class extend the other.

Reply:
Not sure about this. I suppose the helper class is the current attempt to put some code in a common place for both.